package/src/main/java/com/azure/cosmos/cassandra/CosmosRetryPolicy.java

```diff
@@ -0,0 +1,304 @@
+     *
+     * @param driverContext an object holding the context of the current driver instance.
+     * @param profileName   name of the configuration profile to apply.
+     */
+    public CosmosRetryPolicy(final DriverContext driverContext, final String profileName) {
+
+        final DriverExecutionProfile profile = driverContext.getConfig().getProfile(profileName);
+
+        this.maxRetryCount = Option.MAX_RETRIES.getValue(profile);
+        this.fixedBackoffTimeInMillis = Option.FIXED_BACKOFF_TIME.getValue(profile);
+        this.growingBackoffTimeInMillis = Option.GROWING_BACKOFF_TIME.getValue(profile);
+    }
+
+    CosmosRetryPolicy(final int maxRetryCount) {
+        this(maxRetryCount, Option.FIXED_BACKOFF_TIME.getDefaultValue(), Option.GROWING_BACKOFF_TIME.getDefaultValue());
+    }
+
+    // endregion
+
+    // region Accessors
+
+    CosmosRetryPolicy(
+        final int maxRetryCount, final int fixedBackOffTimeInMillis, final int growingBackoffTimeInMillis) {
+
+        this.maxRetryCount = maxRetryCount;
+        this.fixedBackoffTimeInMillis = fixedBackOffTimeInMillis;
+        this.growingBackoffTimeInMillis = growingBackoffTimeInMillis;
+    }
+
+    /**
+     * Gets the {@code fixed-backoff-time} specified by this {@link CosmosRetryPolicy} object.
+     *
+     * @return the {@code fixed-backoff-time} specified by this {@link CosmosRetryPolicy} object.
+     */
+    public int getFixedBackoffTimeInMillis() {
+        return this.fixedBackoffTimeInMillis;
+    }
+
+    /**
+     * Gets the {@code growing-backoff-time} specified by this {@link CosmosRetryPolicy} object.
+     *
+     * @return the {@code growing-backoff-time} specified by this {@link CosmosRetryPolicy} object.
+     */
+    public int getGrowingBackoffTimeInMillis() {
+        return this.growingBackoffTimeInMillis;
+    }
+
+    // endregion
+
+    // region Methods
+
+    /**
+     * Gets the {@code max-retries} specified by this {@link CosmosRetryPolicy} object.
+     *
+     * @return the {@code max-retries} specified by this {@link CosmosRetryPolicy} object.
+     */
+    public int getMaxRetryCount() {
+        return this.maxRetryCount;
+    }
+
+    /**
+     * Closes the current {@link CosmosRetryPolicy}.
+     */
+    @Override
+    public void close() {
+        // nothing to do
+    }
+
+    @Override
+    public RetryDecision onErrorResponse(
+        @NonNull final Request request, @NonNull final CoordinatorException error, final int retryCount) {
+
+        RetryDecision retryDecision;
+
+        // TODO (DANOBLE) ServerError is not the same as ConnectionException and there is no obvious replacement
+        //  Consequently this decision tree must be rethought based on the new and very different CoordinatorException
+        //  hierarchy
+        try {
+            if (error instanceof ServerError) {
+                return this.retryManyTimesOrThrow(retryCount);
```

Comment:
Why do we retry on ServerError?

Reply:
This one needs some discussion. DataStax refactored their exception hierarchy and its unclear to me how some of the exceptions map. I'll leave this as an open issue for now and revisit this before we merge.

Reply:
I removed the ServerError check after confirming that connection exceptions do not pass through retry policy in java-driver-4. Specificaly, when java-driver-4 determines that a contact point host is unknown, or a contact point host refuses a connection, or if connection initialization fails, it will try another contact point. When all contact points are exhausted, java-driver-4 throws an `AllNodesFailedException` to the session consumer. No retry policy code is called. 

This was revealed by code inspection and confirmed by ad-hoc testing. As a result I also removed `CosmosRetryPolicy.canRetryOnConnectionException`.